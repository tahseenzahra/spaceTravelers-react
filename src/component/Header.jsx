import React from 'react';
import { NavLink } from 'react-router-dom';

export default function Header() {
  return (
    <div className="flex justify-between mt-5 shadow-lg pb-5">
      <img className="ml-20 w-[70px] h-[70px]" src="/planet.png" alt="img" />
      <h1 className="ml-5  mt-2 text-4xl font-bold w-full align-middle">Space Travelers Hub</h1>
      <div className="w-full flex justify-end">
        <ul className="flex justify-end w-full gap-5 mt-5 mr-10">
          <NavLink className="text-blue-400 font-bold" to="/">Rockets</NavLink>
          <NavLink className="text-blue-400 font-bold" to="mission">Mission</NavLink>
          <NavLink className="text-blue-400 font-bold" to="dragon">Dragon</NavLink>
          <NavLink className="border-l-2 px-5 h-[70%] border-indigo-500 text-blue-400 font-bold" to="/profile">profile</NavLink>
        </ul>
      </div>
    </div>
  );
}
