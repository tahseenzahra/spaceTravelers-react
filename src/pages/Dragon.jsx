import React, { useState } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import {
  fetchDragons,
  DragonStatus,
} from '../Redux/dragon/dragonSlice';
import Button from '../UI/button';

export default function Dragon() {
  const { dragon } = useSelector((state) => state.dragons);

    const dispatch = useDispatch();
    React.useEffect(() => {
      if (!dragon.length) {
        dispatch(fetchDragons());
      }   
    }, []);
 
  const handeleReserve  = (id) => {
    dispatch(DragonStatus(id))
  }

  return (
    <>
      {dragon.map((dra, i) => (
        <div key={i} className='flex px-20 my-10 mr-60 mx-5'>
          <img className='h-[220px] w-[250px]' src={dra.flickr_images[0]} alt="planet" />
          <div className='ml-5'>
            <h1 className='font-bold mb-3 text-2xl'>{dra.name}</h1>
            <p className='mb-3 text-sm font-serif'> {dra.reservation === false ? <></> :<span className='bg-primary rounded-lg px-2 pb-1 mb-4 mr-2'> reserved </span>} {dra.description}</p>
            <Button className= {dra.reservation === false ? "p-2 bg-primary rounded-lg text-secondary " :'text-secondary rounded-lg border-solid border-2 border-grey-600 p-2'} event = {() => handeleReserve(dra.id)} buttonTag={dra.reservation === false ?'Reserve Dragon' : 'cancelReservation'} />
          </div>
        </div>
      ))}

    </>

  );
}
