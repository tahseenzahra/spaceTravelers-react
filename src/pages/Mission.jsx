import React, { useEffect, useState } from "react";
import { useDispatch, useSelector } from "react-redux";
import Missions from "../UI/missions";
import { loaded } from "../redux/missions/missionSlice";

export default function Mission() {
  const [isJoined, setJoined] = useState("false");
  const dispatch = useDispatch();
  const { mission } = useSelector((state) => state.missions);
  useEffect(() => {
    dispatch(loaded());
  }, []);
  const handleJoin = (e) => {
    mission[0].join_mission = true;
  };
  return (
    <ul
      className="mission-container"
      id="missionList"
    >
      <li className="mission-header">
        <h3 className="mission-column">Mission</h3>
        <h3 className="description-column">Description</h3>
        <h3 className="status-column">Status</h3>
        <div className="join-column" />
      </li>
      {mission.map((mission) => (
        <Missions
          key={Math.random()}
          mission={mission.mission_name}
          description={mission.description}
          status={isJoined ? "not a member" : "Active Member"}
          statusStyle={isJoined ? "member-status inactive-member" : "member-status active-member"}
          buttonTag={isJoined ? "Join Mission" : "Leave Mission"}
          id={mission.mission_id}
          className={isJoined ? "mission-btn join-btn" : "mission-btn leave-btn"}
          event={(e) => {handleJoin(e)}}
        />
      ))}
    </ul>
  );
}
