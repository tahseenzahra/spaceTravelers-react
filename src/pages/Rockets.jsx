import Button from 'react-bootstrap/Button';
import { useEffect } from 'react';
import PropTypes from 'prop-types';
import { useDispatch, useSelector } from 'react-redux';
import { getRockets, reserveRocket } from '../Redux/rockets/rockets';
import './rockets.css';

const Rocket = (props) => {
  const dispatch = useDispatch();
  const { rocket } = props;
  return (
    <div className="each">
      <div className="pictFrame">
        <img
          src={rocket.flickr_images}
          alt={rocket.rocket_name}
          className="pict"
        />
      </div>
      <div className="infor">
        <h2 className="text-3xl">{rocket.rocket_name}</h2>
        <p>
          {rocket.reserved && <span className="reservRock">Reserved</span>}
          {rocket.description}
        </p>
        <Button
          variant={rocket.reserved ? 'outline-primary' : 'primary'}
          type="button"
          className="rocketBtn p-4 bg-primary opacity-50"
          onClick={() => dispatch(reserveRocket(rocket.rocket_id))}
        >
          {rocket.reserved ? 'Cancel reservation' : 'Reserve Rockets'}
        </Button>
      </div>
    </div>
  );
};
Rocket.propTypes = {
  rocket: PropTypes.shape({
    rocket_id: PropTypes.string,
    rocket_name: PropTypes.string,
    description: PropTypes.string,
    reserved: PropTypes.bool,
    flickr_images: PropTypes.arrayOf(PropTypes.string),
  }).isRequired,
};

const Rockets = () => {
  const dispatch = useDispatch();
  const rocketsData = useSelector((state) => state.rockets);
  useEffect(() => {
    if (!rocketsData.length) {
      dispatch(getRockets());
    }
  });
  return (
    <div className="Rocketslist">
      {rocketsData.map((rocketData) => (
        <Rocket key={rocketData.id} rocket={rocketData} id={rocketData.id} />
      ))}
    </div>
  );
};

export default Rockets;
