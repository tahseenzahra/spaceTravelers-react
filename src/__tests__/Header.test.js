import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';

import Header from '../component/Header';
import Rockets from '../pages/Rockets';
import Mission from '../pages/Mission';
import Dragon from '../pages/Dragon';

import store from '../Redux/configureStore';

describe('Header renders correctly', () => {
  test('Renders Header Correctly', () => {
    const header = render(
      <Router>
        <Header />
      </Router>,
    );
    expect(header).toMatchSnapshot();
  });
});

describe('mission renders correctly', () => {
  test('Renders mission Correctly', () => {
    const mission = render(
      <Provider store={store}>
        <Router>
          <Mission />
        </Router>
      </Provider>,
    );
    expect(mission).toMatchSnapshot();
  });
});

describe('rocket renders correctly', () => {
  test('Renders rocket Correctly', () => {
    const rocket = render(
      <Provider store={store}>
        <Router>
          <Rockets />
        </Router>
      </Provider>,
    );
    expect(rocket).toMatchSnapshot();
  });
});

describe('dragon renders correctly', () => {
  test('Renders dragon Correctly', () => {
    const dragon = render(
      <Provider store={store}>
        <Router>
          <Dragon />
        </Router>
      </Provider>,
    );
    expect(dragon).toMatchSnapshot();
  });
});
