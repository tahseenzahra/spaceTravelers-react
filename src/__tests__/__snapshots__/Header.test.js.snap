// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly Renders Header Correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex justify-between mt-5 shadow-lg pb-5"
      >
        <img
          alt="img"
          class="ml-20 w-[70px] h-[70px]"
          src="/planet.png"
        />
        <h1
          class="ml-5  mt-2 text-4xl font-bold w-full align-middle"
        >
          Space Travelers' Hub
        </h1>
        <div
          class="w-full flex justify-end"
        >
          <ul
            class="flex justify-end w-full gap-5 mt-5 mr-10"
          >
            <a
              aria-current="page"
              class="text-primary font-bold active"
              href="/"
            >
              Rockets
            </a>
            <a
              class="text-primary font-bold"
              href="/mission"
            >
              Missions
            </a>
            <a
              class="text-primary font-bold"
              href="/dragon"
            >
              Dragons
            </a>
            <hr
              class="bg-secondary h-6 w-px"
            />
            <a
              class="text-primary font-bold"
              href="/profile"
            >
              My Profile
            </a>
          </ul>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex justify-between mt-5 shadow-lg pb-5"
    >
      <img
        alt="img"
        class="ml-20 w-[70px] h-[70px]"
        src="/planet.png"
      />
      <h1
        class="ml-5  mt-2 text-4xl font-bold w-full align-middle"
      >
        Space Travelers' Hub
      </h1>
      <div
        class="w-full flex justify-end"
      >
        <ul
          class="flex justify-end w-full gap-5 mt-5 mr-10"
        >
          <a
            aria-current="page"
            class="text-primary font-bold active"
            href="/"
          >
            Rockets
          </a>
          <a
            class="text-primary font-bold"
            href="/mission"
          >
            Missions
          </a>
          <a
            class="text-primary font-bold"
            href="/dragon"
          >
            Dragons
          </a>
          <hr
            class="bg-secondary h-6 w-px"
          />
          <a
            class="text-primary font-bold"
            href="/profile"
          >
            My Profile
          </a>
        </ul>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`dragon renders correctly Renders dragon Correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div />
  </body>,
  "container": <div />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`mission renders correctly Renders mission Correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <ul
        class="flex-col mt-4 mx-8"
      >
        <div
          class="kanaderalla"
        >
          Hala
        </div>
        <li
          class="flex justify-between font-bold even:bg-gray-100"
        >
          <h3
            class="w-1/1 p-2 border"
          >
            Mission
          </h3>
          <h3
            class="w-5/7 p-2 border"
          >
            Description
          </h3>
          <h3
            class="w-1/1 p-2 border"
          >
            Status
          </h3>
          <div
            class="w-1/1 border"
          />
        </li>
        <li
          class="flex even:bg-gray-100"
        >
          <div
            class="w-1/1 p-2 border font-bold"
          >
            Thaicom
          </div>
          <div
            class="w-5/7 p-2 border text-sm font-medium"
          >
            Lorem ipsum dolor sit amet consectetur adipisicing elit. Expedita consequatur eum porro neque, cumque quidem nisi molestias dolor dolore assumenda laborum veniam unde ipsa fuga illum aut quo quod maiores.
          </div>
          <div
            class="
 flex
 justify-center
 items-center
 w-1/1
 p-2 border"
          >
            <div
              class="
 bg-secondary
 px-1
 text-font
 uppercase
 place-content-center
 rounded
 font-semibold"
            >
              not a memeber
            </div>
          </div>
          <div
            class="flex justify-center w-1/1 p-2 border"
          >
            <button
              class="bg-transparent border-2 rounded p-2 text-secondary font-semibold text-sm"
              id="id"
              type="button"
            >
              Join Mission
            </button>
          </div>
        </li>
        <li
          class="flex even:bg-gray-100"
        >
          <div
            class="w-1/1 p-2 border font-bold"
          >
            Thaicom
          </div>
          <div
            class="w-5/7 p-2 border text-sm font-medium"
          >
            Lorem ipsum dolor sit amet consectetur adipisicing elit. Expedita consequatur eum porro neque, cumque quidem nisi molestias dolor dolore assumenda laborum veniam unde ipsa fuga illum aut quo quod maiores.
          </div>
          <div
            class="
 flex
 justify-center
 items-center
 w-1/1
 p-2 border"
          >
            <div
              class="
 bg-primary
 px-1
 text-font
 place-content-center
 rounded
 font-semibold"
            >
              Active Member
            </div>
          </div>
          <div
            class="flex justify-center w-1/1 p-2 border"
          >
            <button
              class="bg-transparent border-2 border-alert rounded px-2 py-0 text-alert font-semibold text-sm"
              id="id"
              type="button"
            >
              Leave Mission
            </button>
          </div>
        </li>
      </ul>
    </div>
  </body>,
  "container": <div>
    <ul
      class="flex-col mt-4 mx-8"
    >
      <div
        class="kanaderalla"
      >
        Hala
      </div>
      <li
        class="flex justify-between font-bold even:bg-gray-100"
      >
        <h3
          class="w-1/1 p-2 border"
        >
          Mission
        </h3>
        <h3
          class="w-5/7 p-2 border"
        >
          Description
        </h3>
        <h3
          class="w-1/1 p-2 border"
        >
          Status
        </h3>
        <div
          class="w-1/1 border"
        />
      </li>
      <li
        class="flex even:bg-gray-100"
      >
        <div
          class="w-1/1 p-2 border font-bold"
        >
          Thaicom
        </div>
        <div
          class="w-5/7 p-2 border text-sm font-medium"
        >
          Lorem ipsum dolor sit amet consectetur adipisicing elit. Expedita consequatur eum porro neque, cumque quidem nisi molestias dolor dolore assumenda laborum veniam unde ipsa fuga illum aut quo quod maiores.
        </div>
        <div
          class="
 flex
 justify-center
 items-center
 w-1/1
 p-2 border"
        >
          <div
            class="
 bg-secondary
 px-1
 text-font
 uppercase
 place-content-center
 rounded
 font-semibold"
          >
            not a memeber
          </div>
        </div>
        <div
          class="flex justify-center w-1/1 p-2 border"
        >
          <button
            class="bg-transparent border-2 rounded p-2 text-secondary font-semibold text-sm"
            id="id"
            type="button"
          >
            Join Mission
          </button>
        </div>
      </li>
      <li
        class="flex even:bg-gray-100"
      >
        <div
          class="w-1/1 p-2 border font-bold"
        >
          Thaicom
        </div>
        <div
          class="w-5/7 p-2 border text-sm font-medium"
        >
          Lorem ipsum dolor sit amet consectetur adipisicing elit. Expedita consequatur eum porro neque, cumque quidem nisi molestias dolor dolore assumenda laborum veniam unde ipsa fuga illum aut quo quod maiores.
        </div>
        <div
          class="
 flex
 justify-center
 items-center
 w-1/1
 p-2 border"
        >
          <div
            class="
 bg-primary
 px-1
 text-font
 place-content-center
 rounded
 font-semibold"
          >
            Active Member
          </div>
        </div>
        <div
          class="flex justify-center w-1/1 p-2 border"
        >
          <button
            class="bg-transparent border-2 border-alert rounded px-2 py-0 text-alert font-semibold text-sm"
            id="id"
            type="button"
          >
            Leave Mission
          </button>
        </div>
      </li>
    </ul>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`rocket renders correctly Renders rocket Correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <h1>
          Rockets
        </h1>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <h1>
        Rockets
      </h1>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
